### PR TITLE
convert tests to json format to make generate file nicer

### PR DESCRIPTION
**Pull request by Kuba314**

### CODE REVIEW

File: tests.json [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    "delim": ","
+  },
+  {
+    "name": "Calculates 25th fibonacci number. Requires implemented goto's and relevant.",
+    "output": "fib25",
+    "input": "empty_input",
+    "cmds": "set 25;[1,2];set 1;[1,3];set 1;[1,1];def _4;[1,2];def _1;[1,3];def _2;[1,1];set 1;def _0;sub _4 _0;iszero _4 +15;set 0;use _1;def _3;set 0;def _0;sub _0 _2;sub _3 _0;set 0;use _2;def _1;set 0;use _3;def _2;goto -17;use _1",
+    "delim": ":"
+  }
+]
```
Comment (Adrijaned) [inline tests.json:192]: newline

File: generate_reference_solutions.py [hunk 2]
```diff
@@ -24,19 +25,20 @@ def run_test(self):
...
+
+        test_cases += [TestCase(args, test['input'], test['name'], test['output'])]
 
     for test_case in test_cases:
         test_case.run_test()
 
 
 if __name__ == '__main__':
-    main()
+    main()
```
Comment (Adrijaned) [inline generate_reference_solutions.py]: All files should end in newline (see POSIX https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206 for reasons; while this is especially true for programs targeted on Linux such as this, it is customary to maintain this convention whenever possible)

**Review by Adrijaned — CHANGES_REQUESTED** [review]
Taking the JSON path is definitely the "good" way, although, as you might have noticed, it complicates some quoted/backslashed strings quite a bit, which is after all the reason I didn't go fot it in the first place. But most tests I feel are needed using those have already been written, so since you've taken the time to do those, I don't see this as an issue anymore. So, just two details:

- fix those two newlines please
- get rid of the old `tests` file

**Comment by Kuba314** [comment]
Added newline and removed the old `tests` file

**Review by Adrijaned — APPROVED** [review]
JSON still has missing newline at the end, but that will probably resolve itself once anyone adds any more tests :)

**Pull request merged**